h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -3366,16 +3384,75 @@ private void scorePostProcessing(Frame train, long t1) {
       } else {
         Log.info(LogMsg("ModelMetrics mtrain is null"));
       }
-      Log.info(LogMsg("Training metrics computed in " + (t2 - t1) + "ms"));
+      Log.info(LogMsg("Control values training metrics computed in " + (t2 - t1) + "ms"));
       if (_valid != null) {
         Frame valid = DKV.<Frame>getGet(_parms._valid);
+        _model._useControlVariables = true;
         _model.score(_parms.valid(), null, CFuncRef.from(_parms._custom_metric_func)).delete();
+        _model._useControlVariables = false;
```

Comment:
Maybe use try/finally?

I think here might be possible for custom metric function to misbehave and throw an exception causing the model to have `_useControlVariables=true`.